nvflare/private/fed/app/client/client_train.py

```diff
@@ -171,6 +172,13 @@ def create_admin_agent(
     return admin_agent
 
 
+def _shutdown_daemon_process(workspace: Workspace):
+    # touch files to trigger shutdown of daemon process
+    name = workspace.get_file_path_in_root(WorkspaceConstants.SHUTDOWN_FILE)
+    print(f"creating file {name}")
```

Comment:
This will rely on the start script to use SHUTDOWN_FILE to kill the running process. We should not rely on the external action to terminate the program, instead if should terminate itself. Otherwise, if the program is not started by the start.sh script, then it won't work.

Reply:
This should be handled already by MPM shutdown. 

What i do here is to make sure start.sh don't launch the client process again in short period.